### ISSUE TITLE
CA1508 does not respect implicit casts

### ISSUE THREAD

**Issue by hankovich** [dotnet/roslyn-analyzers#6823]
### Analyzer

**Diagnostic ID**: [CA1508](https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1508): Avoid dead conditional code

### Analyzer source

**NuGet Package**: [Microsoft.CodeAnalysis.NetAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.NetAnalyzers)

**Version**: 8.0.0-preview1.23165.1 (Latest)

### Steps To Reproduce

```csharp
var redisValue = RedisValue.Null; // From StackExchange.Redis. RedisValue has an implicit cast to byte[]?
var blobValue = (byte[]?)redisValue;

if (blobValue == null) // CA1508 `blobValue == null` is always `false`
{
    return;
}

Console.WriteLine("wasted");
```

### Expected behavior

No diagnostic is reported.

### Actual behavior

CA1508 diagnostic is reported.

**Comment by hankovich** [comment]
Closing as a duplicate of https://github.com/dotnet/roslyn-analyzers/issues/6435
